test/system/252-quadlet.bats

```diff
@@ -0,0 +1,251 @@
+#!/usr/bin/env bats   -*- bats -*-
+#
+# Tests generated configurations for systemd.
+#
+
+load helpers
+load helpers.systemd
+
+UNIT_FILES=()
+
+function start_time() {
+    sleep_to_next_second # Ensure we're on a new second with no previous logging
+    STARTED_TIME=$(date "+%F %R:%S") # Start time for new log time
+}
+
+function setup() {
+    skip_if_remote "quadlet tests are meaningless over remote"
+
+    start_time
+
+    basic_setup
+}
+
+function teardown() {
+    for UNIT_FILE in ${UNIT_FILES[@]}; do
+        if [[ -e "$UNIT_FILE" ]]; then
+            local service=$(basename "$UNIT_FILE")
+            run systemctl stop "$service"
+            if [ $status -ne 0 ]; then
+               echo "# WARNING: systemctl stop failed in teardown: $output" >&3
+            fi
+            rm -f "$UNIT_FILE"
+        fi
+    done
+    systemctl daemon-reload
+
+    basic_teardown
+}
+
+# Converts the quadlet file and installs the result it in $UNIT_DIR
+function run_quadlet() {
+    local sourcefile="$1"
+    local service=$(quadlet_to_service_name "$sourcefile")
+
+    # quadlet always works on an entire directory, so copy the file
+    # to transform to a tmpdir
+    local quadlet_tmpdir=$(mktemp -d --tmpdir=$PODMAN_TMPDIR quadlet.XXXXXX)
```

Comment:
Can we `rm -rf $quadlet_tmpdir` after?

Reply:
I tried that first, but it caused systemd to report some error, probably because the source the generated file points to is gone.